packages/sanity/src/core/config/types.ts

```diff
@@ -671,6 +681,14 @@ export interface Source {
     comments: {
       enabled: (props: DocumentCommentsEnabledContext) => boolean
     }
+
+    /**
+     * @internal
+     * @hidden
+     */
+    unstable_legacyArrayEditing?: {
```

Comment:
I think it would make more sense to have this property in `form` rather than in `document`. There is already an `unstable` property in `form` (`form.unstable.<thing>`) where this property could potentially be added to!

What do you think? :) 

```ts
export default defineConfig({
  // ...

  form: {
    unstable: {
      legacyArrayEditing: true,
    },
  },
})
```

Reply:
I'm not married to the approach, the first time I did think about putting it in form. I thought about document because I felt it might be something that impacts outside of the form just itself (mainly considering that the "disable" of actions is outside of the form though it's impacted about the editing dialog).

It made more sense to me to keep it in document because of that. What do you think? Keep it in document or move it to form.unstable?

Reply:
In this case, I think it makes more sense to have this property in `form`, as that’s where the current configuration of the form is happening (such as inputs, fields, files, images, etc.). I'd say that the `document` property is more about configuring things _around_ the form, such as actions, badges, inspectors, comments, etc., whereas the `form` property is more focused on configuring the actual form itself. Since the new property we are adding is about configuring the array input in the form, I believe having the property in `form` rather than in the `document` is more correct!

Happy to discuss this further tomorrow in a meeting :)

Reply:
Resolving this, after talking with Rune he suggested a different approach that aligns better with what DX wants to do as well. 
I've just pushed a commit with it, let's talk on our meeting :)